docs/src/dev-docs/design-docker-compose.md

```
@@ -0,0 +1,101 @@
+Docker Compose manages service startup order through:
+
+* `depends_on` directives.
+* Health checks with `condition: service_healthy`.
+* Init containers for one-time setup tasks (e.g., `db-table-creator`).
+
+## Troubleshooting
+
+If you encounter issues with the Docker Compose deployment:
+
+1. Check service status:
+   ```bash
```

markdownlint-cli2 (0.17.2)

Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)